src-generate/Generator.hs

```diff
@@ -461,6 +462,19 @@ getInsertFunc maxCapability typeDesc = unlines ["{-# INLINE " ++ funcName ++ " #
         toCasesHelper (_, x) [] = ["| otherwise " ++ x]
         toCasesHelper (c, x) xs = ("| " ++ c ++ " " ++ x) : xs
 
+-- | Generate a function to generate a vector from an index-function
+getGenerateFunc :: TypeDesc -> String
+getGenerateFunc typeDesc = unlines
+    ["{-# INLINE[1] " ++ funcName ++ " #-}"
```

Comment:
I'm not sure if `INLINE[1]` is the best choice here. I adapted that from `mapVector`. After all, `generateVector` also maps a function, just not over elements of an already-given vector but over ascending indices.

Also, I don't know if it would be useful to implement this function, like `mapVector`, in terms of an unboxed version `generateVector#`.